extensions/wikia/AdEngine/js/spec/lookup/rubiconVulcan.spec.js

```diff
@@ -0,0 +1,181 @@
+/*global beforeEach, describe, it, modules, expect, spyOn*/
+describe('ext.wikia.adEngine.lookup.rubiconVulcan', function () {
+	'use strict';
+	function noop() {
+	}
+
+	var mocks = {
+			adContext: {
+				getContext: function () {
+					return {
+						opts: mocks.opts,
+						slots: noop,
+						targeting: mocks.targeting
+					};
+				}
+			},
+			tracker: {
+				measureDiff: function () {
+					return mocks.tracker;
+				},
+				track: noop
+			},
+			adTracker: {
+				measureTime: function () {
+					return mocks.tracker;
+				},
+				track: noop
+			},
+			doc: {
+				node: {
+					addEventListener: function (ignore, callback) {
+						callback();
+					},
+					parentNode: {
+						insertBefore: noop
+					}
+				},
+				createElement: function () {
+					return mocks.doc.node;
+				},
+				getElementsByTagName: function () {
+					return [mocks.doc.node];
+				}
+			},
+			lazyQueue: {
+				makeQueue: function (queue, callback) {
+					queue.push = function () {
+						callback();
+					};
+					queue.start = noop;
+				}
+			},
+			log: noop,
+			recoveryHelper: {
+				addOnBlockingCallback: noop
+			},
+			rubiconTargeting: {
+				getTargeting: function () {
+					return {
+						lang: 'en',
+						passback: 'vulcan',
+						s0: 'life',
+						s1: '_dragonball',
+						s2: 'article',
+						src: 'gpt'
+					};
+				}
+			},
+			slot: {
+				id: 'INCONTENT_LEADERBOARD',
+				getBestCpm: function () {
+					return {
+						cpm: 16.2343446,
+						status: 'ok',
+						type: 'vast'
+					};
+				}
+			},
+			targeting: {
+				skin: 'oasis'
+			},
+			tiers: [],
+			win: {
+				rubicontag: {
+					video: {
+						run: function (onResponse) {
+							onResponse();
+						},
+						defineSlot: function () {
+							return mocks.slot;
+						},
+						getAllSlots: function () {
+							return [mocks.slot];
+						}
+					}
+
+				}
+			}
+		};
+
+	function getFactory() {
+		return modules['ext.wikia.adEngine.lookup.lookupFactory'](
+			mocks.adContext,
+			mocks.adTracker,
+			mocks.recoveryHelper,
+			mocks.lazyQueue,
+			mocks.log
+		);
+	}
+
+	function getVulcan() {
+		return modules['ext.wikia.adEngine.lookup.rubiconVulcan'](
+			getFactory(),
+			mocks.rubiconTargeting,
+			modules['ext.wikia.adEngine.utils.math'](),
+			mocks.doc,
+			mocks.log,
+			mocks.win
+		);
+	}
+	
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3ATrailingWhitespace)